x-pack/plugins/timelines/public/store/t_grid/helpers.ts

```diff
@@ -160,20 +160,24 @@ export const setInitializeTgridSettings = ({
 }: InitializeTgridParams): TimelineById => {
   const timeline = timelineById[id];
 
-  return {
-    ...timelineById,
-    [id]: {
-      ...tGridDefaults,
-      ...timeline,
-      ...getTGridManageDefaults(id),
-      ...tGridSettingsProps,
-      ...(!timeline || (isEmpty(timeline.columns) && !isEmpty(tGridSettingsProps.defaultColumns))
-        ? { columns: tGridSettingsProps.defaultColumns }
-        : {}),
-      sort: tGridSettingsProps.sort ?? tGridDefaults.sort,
-      loadingEventIds: tGridDefaults.loadingEventIds,
-    },
-  };
+  return !timeline?.initialized
+    ? {
+        ...timelineById,
+        [id]: {
+          ...tGridDefaults,
+          ...timeline,
+          ...getTGridManageDefaults(id),
```

Comment:
I think all you have to do to fix this bug is spread `timeline` underneath `getTGridManageDefaults(id)`

Reply:
Hey @stephmilovic, thank you for taking a look at this PR! 

I also thought the same way. So I even pushed this [commit](https://github.com/elastic/kibana/commit/15b8e3eb01c89c0e4ae4c59ae0333a650e17ee02#diff-8722f79b4ed144ce22175c7374dbba141e719ac7974bd9df9e8191f41ee2e77aL163) with the fix you mentioned, and it does solve the timeline name issue.
But it doesn't fix #106819 issue (I am not sure if event rendering is still relevant after t-grid). Because `excludedRowRendererIds` is defined inside `tGridSettingsProps` which is also spread after `timeline`. 


After that, I  had a conversation with @XavierM where he mentioned some concerns about this solution and suggested the redux store flag approach. I can't remember which were his concerns 🤦‍♂️ . @XavierM, do you remember it?

Reply:
ok i figured there was another bug i wasn't checking. ill remove my change request. thanks!

Reply:
i feel like we should still move `timeline` to be spread beneath `getTGridManageDefaults`. dont want to lose any properties and overwrite them with defaults, right?

Reply:
I agreed with you. @XavierM, do you know any reason why we shouldn't do it?